lib/puppet/provider/cs_property/crm.rb

```diff
@@ -81,7 +81,37 @@ def flush
       # clear this on properties, in case it's set from a previous
       # run of a different corosync type
       ENV['CIB_shadow'] = nil
-      crm('configure', 'property', '$id="cib-bootstrap-options"', "#{@property_hash[:name]}=#{@property_hash[:value]}")
+      begin
+        crm('configure', 'property', '$id="cib-bootstrap-options"', "#{@property_hash[:name]}=#{@property_hash[:value]}")
+      rescue
+        xml = <<-EOF
+        <diff>
+          <diff-removed>
+            <cib>
+              <configuration>
+                <crm_config>
+                  <cluster_property_set id="cib-bootstrap-options">
+                    <nvpair value=#{@property_hash[:value].inspect} id=#{@property_hash[:name].inspect}/>
+                  </cluster_property_set>
+                </crm_config>
+              </configuration>
+            </cib>
+          </diff-removed>
+          <diff-added>
+            <cib>
+              <configuration>
+                <crm_config>
+                  <cluster_property_set id="cib-bootstrap-options">
+                    <nvpair value="stop" id="cib-bootstrap-options-no-quorum-policy"/>
+                  </cluster_property_set>
+                </crm_config>
+              </configuration>
+            </cib>
+          </diff-added>
+        </diff>
```

Comment:
this code makes me sad
i mean, xml in general makes me sad, but a giant xml string in the middle of our code makes me even saddener.
is there a way to make a computer generate this?